src/sentry/migrations/0707_alert_rule_activations_incidents_fk.py

```diff
@@ -0,0 +1,73 @@
+# Generated by Django 5.0.3 on 2024-04-24 23:38
+
+import django.db.models.deletion
+from django.db import migrations, models
+
+import sentry.db.models.fields.foreignkey
+from sentry.new_migrations.migrations import CheckedMigration
+
+
+class Migration(CheckedMigration):
+    # This flag is used to mark that a migration shouldn't be automatically run in production.
+    # This should only be used for operations where it's safe to run the migration after your
+    # code has deployed. So this should not be used for most operations that alter the schema
+    # of a table.
+    # Here are some things that make sense to mark as post deployment:
+    # - Large data migrations. Typically we want these to be run manually so that they can be
+    #   monitored and not block the deploy for a long period of time while they run.
+    # - Adding indexes to large tables. Since this can take a long time, we'd generally prefer to
+    #   run this outside deployments so that we don't block them. Note that while adding an index
+    #   is a schema change, it's completely safe to run the operation after the code has deployed.
+    # Once deployed, run these manually via: https://develop.sentry.dev/database-migrations/#migration-deployment
+
+    is_post_deployment = False
+
+    dependencies = [
+        ("sentry", "0706_grouphistory_userteam_backfill"),
+    ]
+
+    operations = [
+        migrations.SeparateDatabaseAndState(
+            database_operations=[
+                migrations.RunSQL(
+                    """
+                    ALTER TABLE "sentry_alertruleactivations" ADD COLUMN "activator" VARCHAR(250) NOT NULL DEFAULT 'default_activator';
+                    """,
+                    reverse_sql="""
+                    ALTER TABLE "sentry_alertruleactivations" DROP COLUMN "activator";
+                    """,
+                    hints={"tables": ["sentry_alertruleactivations"]},
+                ),
+                migrations.RunSQL(
+                    """
+                    ALTER TABLE "sentry_alertruleactivations" ADD COLUMN "condition_type" SMALLINT NOT NULL DEFAULT 0;
+                    """,
+                    reverse_sql="""
+                    ALTER TABLE "sentry_alertruleactivations" DROP COLUMN "condition_type";
+                    """,
+                    hints={"tables": ["sentry_alertruleactivations"]},
+                ),
+            ],
+            state_operations=[
+                migrations.AddField(
+                    model_name="alertruleactivations",
+                    name="activator",
+                    field=models.CharField(max_length=250, default="default_activator"),
```

Comment:
nit: for the future it might be nice if this was a json blob -- likely fine to leave as a string for the `ids` though.